src/esp/assets/PTexMeshData.cpp

```diff
@@ -220,12 +247,116 @@ std::vector<PTexMeshData::MeshData> splitMesh(
       uint32_t index = refdVerts[j];
       subMeshes[i].vbo[j] = mesh.vbo[index];
       subMeshes[i].nbo[j] = mesh.nbo[index];
+      // Careful:
+      // for Ptex mesh we never ever set the "cbo"
     }
   }
 
   return subMeshes;
 }
 
+// =========== the input file format =======================
+// a uint64_t, N, the number of sub-meshes;
+
+// it follows by N chunks, each of which contains:
+// a uint64_t, M, the number of faces within the chunk;
+// M uint32_t, face indices (sorted) in the *original* mesh;
+// =========================================================
+
+//  this binary can be generated by modifying the splitMesh() in file
+//  PTexLib.cpp in ReplicaSDK.
+//  and the name is hard-coded as "sorted_faces.bin" in the simulator.
+
+// Put it in the sub-folder, "habitat".
+
+std::vector<PTexMeshData::MeshData> loadSubMeshes(
+    const PTexMeshData::MeshData& mesh,
+    const std::string& filename) {
+  // sanity checks
+  CORRADE_ASSERT(!filename.empty(), "Error: filename cannot be empty.", {});
+  std::ifstream file;
+  file.open(filename, std::ios::in | std::ios::binary);
+  CORRADE_ASSERT(file.good(), "Error: cannot open the file " << filename, {});
+
+  uint64_t numSubMeshes = 0;
+  file.read((char*)&numSubMeshes, sizeof(uint64_t));
```

Comment:
static_cast<char *>

Reply:
OK, will do.

Reply:
Oh, actually I suddenly remembered that: 
 static_cast from 'uint64_t *' (aka 'unsigned long long *') to 'char *' is not allowed.

Just tried and double confirmed it.

Reply:
Ah. reinterpret_cast<uint64_t *> then to be consistent with the style guide:

https://google.github.io/styleguide/cppguide.html#Casting